### PR TITLE
chore: CE: move action, jsObject datatree methods

### PR DESCRIPTION
**Pull request by dvj1988**
Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags=""

### :mag: Cypress test results

> [!CAUTION]  
> If you modify the content in this section, you are likely to disrupt the CI result for your PR.

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

### CODE REVIEW

File: app/client/packages/types/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "@appsmith/types",
```
Comment (znamenskii-ilia) [inline app/client/packages/types/package.json:2]: I'm curios to know what is the reason for creating this packages?

File: app/client/packages/evaluation/src/dataTree/action/index.ts [hunk 2]
```diff
@@ -254,48 +259,132 @@ export const getBindingAndReactivePathsOfAction = (
...
+    isLoading: boolean;
+    config: {
+      id: string;
+      name: string;
+      pluginId: string;
+      dynamicBindingPathList: { key: string; value?: string }[];
+      datasource: { datasourceConfiguration?: { url?: string } };
+      // TODO: Fix this the next time the file is edited
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      actionConfiguration: any;
```
Comment (rishabhrathod01) [inline app/client/packages/evaluation/src/dataTree/action/index.ts:276]: Avoid adding any and define type instead.

File: app/client/packages/evaluation/src/dataTree/action/index.ts [hunk 3]
```diff
@@ -254,48 +259,132 @@ export const getBindingAndReactivePathsOfAction = (
...
+    };
+    data?: {
+      statusCode: string;
+      isExecutionSuccess?: boolean;
+      headers: Record<string, string[]>;
+    };
+  },
+  // TODO: Fix this the next time the file is edited
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  editorConfig: any[],
```
Comment (rishabhrathod01) [inline app/client/packages/evaluation/src/dataTree/action/index.ts:287]: Avoid using any

File: app/client/packages/evaluation/src/dataTree/ce/moduleInstance.ts [hunk 4]
```diff
@@ -1,11 +1,11 @@
-import type { ModuleInstance } from "ee/constants/ModuleInstanceConstants";
-
 export const generateModuleInstance = (
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  moduleInstance: ModuleInstance,
+  moduleInstance: {
```
Comment (rishabhrathod01) [inline app/client/packages/evaluation/src/dataTree/ce/moduleInstance.ts:3]: Type and the name of variable doesn't seem to match here.

**Pull request closed without merging**